src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -123,11 +127,18 @@ protected override async Task LoadComponentsAsync(CancellationToken cancellation
             // the appropriate task scheduler to report events on.
             this.ComponentModel.GetService<MiscellaneousFilesWorkspace>();
 
-            LoadAnalyzerNodeComponents();
+            LoadAnalyzerNodeComponents();            
 
             LoadComponentsBackgroundAsync(cancellationToken).Forget();
         }
 
+        private void RegisterCodeCleanupProvider()
+        {
+            var regService = ComponentModel.GetService<ICodeCleanUpFixerRegistrationService>();
+            var provider = ComponentModel.GetService<CodeCleanUpFixerProvider>();
```

Comment:
is there a reason we're importing concrete classes, instead of just interfaces?

Reply:
we don't have interface except the one from editor. and interface from editor is not supposed to be used for MEF since it is not something we owned.

Reply:
There is a ICodeCleanUpFixerProvider interface, but I think you are wanting to import your specific provider rather than any provider.

Reply:
usually, one is not supposed to MEF export type/interface it doesn't own unless that type/interface is explicitly designed to be used in MEF 

otherwise, host (VS in this case) can fail randomly since someone is MEF exporting type that is not supposed to MEF export that get rolled into host MEF composition.


example will be, if some random one starts to MEF export VSWorkspace, VS will just fail.